frozendict/frozendict.pyi

```diff
@@ -1,6 +1,57 @@
-try:
-    from ._frozendict import *
-except ImportError:
-    from .core import *
+import collections
+from _typeshed import Self
+from collections.abc import Iterable, Iterator, Mapping
+from typing import Any, Generic, NoReturn, Tuple, TypeVar, Union, overload
 
-FrozenOrderedDict = frozendict
+_KT = TypeVar("_KT")
+_VT = TypeVar("_VT")
+
+def frozendict_or(
+    self: Mapping[_KT, _VT], other: Mapping[_KT, _VT]
+) -> "frozendict[_KT, _VT]": ...
+
+class frozendict(Mapping[_KT, _VT], Generic[_KT, _VT]):
```

Comment:
Why Generic?

Reply:
It tells the type system that the class itself supports two generic types.

Reply:
But `Generic[T1, T2]` means the class is generic and supports two parameters. This is not true, since the class is a map. There's also an example of custom `dict` on `typing` official docs, and it uses `Mapping`, not `Generic`.